File: client_libraries/javascript/fist-client.js
/**
 * FIST Content Moderation JavaScript Client Library
 * 
 * A JavaScript/Node.js client library for easy integration with the FIST Content Moderation API.
 * Provides simple methods for content moderation, batch processing, and monitoring.
 * 
 * Usage:
 *   // Node.js
 *   const { FistClient } = require('./fist-client');
 *   
 *   // ES6 Modules / Browser
 *   import { FistClient } from './fist-client.js';
 *   
 *   const client = new FistClient({
 *     token: 'fist_your_token_here',
 *     baseUrl: 'https://your-fist-api.com'
 *   });
 *   
 *   // Single content moderation
 *   const result = await client.moderateContent('Some text to moderate');
 *   
 *   // Batch processing
 *   const results = await client.moderateBatch(['text1', 'text2', 'text3']);
 */

// Detect environment (Node.js vs Browser)
const isNode = typeof window === 'undefined' && typeof global !== 'undefined';
let fetch;

if (isNode) {
    // Node.js environment - try to import node-fetch
    try {
        fetch = require('node-fetch');
    } catch (e) {
        console.warn('node-fetch not found. Please install it: npm install node-fetch');
        throw new Error('node-fetch is required for Node.js environments');
    }
} else {
    // Browser environment - use native fetch
    fetch = window.fetch;
}

/**
 * Custom error classes for FIST client
 */
class FistClientError extends Error {
    constructor(message) {
        super(message);
        this.name = 'FistClientError';
    }
}

class AuthenticationError extends FistClientError {
    constructor(message = 'Authentication failed') {
        super(message);
        this.name = 'AuthenticationError';
    }
}

class APIError extends FistClientError {
    constructor(message, statusCode = null, responseData = null) {
        super(message);
        this.name = 'APIError';
        this.statusCode = statusCode;
        this.responseData = responseData;
    }
}

/**
 * Data classes for structured responses
 */
class ModerationResult {
    constructor(data) {
        this.moderationId = data.moderation_id;
        this.contentHash = data.content_hash;
        this.inappropriateProbability = data.ai_result.inappropriate_probability;
        this.aiReason = data.ai_result.reason;
        this.finalDecision = data.final_decision;
        this.reason = data.reason;
        this.wordCount = data.word_count;
        this.percentageUsed = data.percentage_used;
        this.createdAt = data.created_at;
        this.fromCache = data.from_cache || false;
    }
}

class BatchJob {
    constructor(data) {
        this.jobId = data.job_id;
        this.status = data.status;
        this.totalItems = data.total_items;
        this.processedItems = data.processed_items;
        this.progressPercent = data.progress_percent;
        this.createdAt = data.created_at;
        this.startedAt = data.started_at || null;
        this.completedAt = data.completed_at || null;
        this.backgroundTaskId = data.background_task_id || null;
    }
}

/**
 * Main FIST Client class
 */
class FistClient {
    /**
     * Initialize FIST client
     * 
     * @param {Object} options - Configuration options
     * @param {string} options.token - API token (should start with 'fist_')
     * @param {string} [options.baseUrl='http://localhost:8000'] - Base URL of the FIST API
     * @param {number} [options.timeout=30000] - Request timeout in milliseconds
     * @param {number} [options.maxRetries=3] - Maximum number of retries for failed requests
     * @param {number} [options.retryDelay=1000] - Delay between retries in milliseconds
     */
    constructor(options = {}) {
        const {
            token,
            baseUrl = 'http://localhost:8000',
            timeout = 30000,
            maxRetries = 3,
            retryDelay = 1000
        } = options;

        if (!token) {
            throw new Error('Token is required');
        }

        if (!token.startsWith('fist_')) {
            throw new Error('Token must start with "fist_"');
        }

        this.token = token;
        this.baseUrl = baseUrl.replace(/\/$/, ''); // Remove trailing slash
        this.timeout = timeout;
        this.maxRetries = maxRetries;
        this.retryDelay = retryDelay;

        this.defaultHeaders = {
            'Authorization': `Bearer ${token}`,
            'Content-Type': 'application/json',
            'User-Agent': 'FIST-JavaScript-Client/1.0.0'
        };
    }

    /**
     * Make HTTP request with retry logic
     * 
     * @private
     * @param {string} method - HTTP method
     * @param {string} endpoint - API endpoint
     * @param {Object} [data] - Request body data
     * @param {Object} [params] - URL parameters
     * @returns {Promise<Object>} Response data
     */
    async _makeRequest(method, endpoint, data = null, params = null) {
        const url = new URL(endpoint, this.baseUrl);
        
        if (params) {
            Object.keys(params).forEach(key => {
                url.searchParams.append(key, params[key]);
            });
        }

        const requestOptions = {
            method: method.toUpperCase(),
            headers: { ...this.defaultHeaders },
            timeout: this.timeout
        };

        if (data && ['POST', 'PUT', 'PATCH'].includes(method.toUpperCase())) {
            requestOptions.body = JSON.stringify(data);
        }

        for (let attempt = 0; attempt <= this.maxRetries; attempt++) {
            try {
                const response = await fetch(url.toString(), requestOptions);

                // Handle authentication errors
                if (response.status === 401) {
                    throw new AuthenticationError('Invalid or expired token');
                }

                // Handle other client/server errors
                if (response.status >= 400) {
                    let errorMessage;
                    let errorData = null;

                    try {
                        errorData = await response.json();
                        errorMessage = errorData.detail || `HTTP ${response.status}`;
                    } catch {
                        errorMessage = `HTTP ${response.status}: ${response.statusText}`;
                    }

                    throw new APIError(errorMessage, response.status, errorData);
                }

                return await response.json();

            } catch (error) {
                if (attempt === this.maxRetries) {
                    throw error;
                }

                // Don't retry client errors (4xx)
                if (error instanceof APIError && error.statusCode < 500) {
                    throw error;
                }

                // Exponential backoff
                await this._sleep(this.retryDelay * Math.pow(2, attempt));
            }
        }

        throw new APIError('Max retries exceeded');
    }

    /**
     * Sleep utility function
     * 
     * @private
     * @param {number} ms - Milliseconds to sleep
     * @returns {Promise<void>}
     */
    _sleep(ms) {
        return new Promise(resolve => setTimeout(resolve, ms));
    }

    /**
     * Moderate a single piece of content
     * 
     * @param {string} content - Text content to moderate
     * @param {Object} [options] - Moderation options
     * @param {number[]} [options.percentages] - Custom percentages for content piercing
     * @param {number[]} [options.thresholds] - Custom word count thresholds
     * @param {Object} [options.probabilityThresholds] - Custom probability thresholds
     * @returns {Promise<ModerationResult>} Moderation result
     */
    async moderateContent(content, options = {}) {
        if (!content || !content.trim()) {
            throw new Error('Content cannot be empty');
        }

        const data = { content };
        
        if (options.percentages) data.percentages = options.percentages;
        if (options.thresholds) data.thresholds = options.thresholds;
        if (options.probabilityThresholds) data.probability_thresholds = options.probabilityThresholds;

        const response = await this._makeRequest('POST', '/api/moderate', data);
        return new ModerationResult(response.result);
    }

    /**
     * Moderate multiple pieces of content in batch
     * 
     * @param {string[]} contents - Array of text content to moderate
     * @param {Object} [options] - Batch processing options
     * @param {number[]} [options.percentages] - Custom percentages for content piercing
     * @param {number[]} [options.thresholds] - Custom word count thresholds
     * @param {Object} [options.probabilityThresholds] - Custom probability thresholds
     * @param {boolean} [options.background=false] - Process in background
     * @param {number} [options.pollInterval=2000] - Polling interval for background jobs (ms)
     * @param {number} [options.maxWaitTime=300000] - Maximum time to wait for background jobs (ms)
     * @returns {Promise<ModerationResult[]|BatchJob>} Array of results or BatchJob object
     */
    async moderateBatch(contents, options = {}) {
        if (!contents || contents.length === 0) {
            throw new Error('Contents array cannot be empty');
        }

        if (contents.length > 100) {
            throw new Error('Batch size cannot exceed 100 items');
        }

        const {
            percentages,
            thresholds,
            probabilityThresholds,
            background = false,
            pollInterval = 2000,
            maxWaitTime = 300000
        } = options;

        const data = {
            contents,
            background
        };

        if (percentages) data.percentages = percentages;
        if (thresholds) data.thresholds = thresholds;
        if (probabilityThresholds) data.probability_thresholds = probabilityThresholds;

        const response = await this._makeRequest('POST', '/api/moderate/batch', data);

        if (background) {
            const batchJob = new BatchJob(response);
            
            // Wait for completion if requested
            if (maxWaitTime > 0) {
                return await this._waitForBatchCompletion(batchJob.jobId, pollInterval, maxWaitTime);
            }
            
            return batchJob;
        } else {
            // Return results directly
            return (response.results || []).map(result => new ModerationResult(result));
        }
    }

    /**
     * Get status of a batch processing job
     * 
     * @param {string} jobId - Batch job ID
     * @returns {Promise<BatchJob>} Batch job status
     */
    async getBatchStatus(jobId) {
        const response = await this._makeRequest('GET', `/api/moderate/batch/${jobId}/status`);
        return new BatchJob(response);
    }

    /**
     * Get results of a completed batch job
     * 
     * @param {string} jobId - Batch job ID
     * @returns {Promise<ModerationResult[]>} Array of moderation results
     */
    async getBatchResults(jobId) {
        const response = await this._makeRequest('GET', `/api/moderate/batch/${jobId}/results`);
        return (response.results || []).map(result => new ModerationResult(result));
    }

    /**
     * Wait for batch job completion and return results
     * 
     * @private
     * @param {string} jobId - Batch job ID
     * @param {number} pollInterval - Polling interval in milliseconds
     * @param {number} maxWaitTime - Maximum wait time in milliseconds
     * @returns {Promise<ModerationResult[]>} Array of moderation results
     */
    async _waitForBatchCompletion(jobId, pollInterval, maxWaitTime) {
        const startTime = Date.now();

        while (Date.now() - startTime < maxWaitTime) {
            const status = await this.getBatchStatus(jobId);

            if (status.status === 'completed') {
                return await this.getBatchResults(jobId);
            } else if (status.status === 'failed') {
                throw new APIError(`Batch job ${jobId} failed`);
            }

            await this._sleep(pollInterval);
        }

        throw new APIError(`Batch job ${jobId} did not complete within ${maxWaitTime}ms`);
    }

    /**
     * Get system health status
     * 
     * @returns {Promise<Object>} Health check results
     */
    async getHealth() {
        return await this._makeRequest('GET', '/api/health');
    }

    /**
     * Get system performance metrics
     * 
     * @returns {Promise<Object>} Performance metrics
     */
    async getMetrics() {
        return await this._makeRequest('GET', '/api/metrics');
    }

    /**
     * Get cache performance statistics
     * 
     * @returns {Promise<Object>} Cache statistics
     */
    async getCacheStats() {
        return await this._makeRequest('GET', '/api/cache/stats');
    }

    /**
     * Clear the system cache
     * 
     * @returns {Promise<Object>} Clear operation results
     */
    async clearCache() {
        return await this._makeRequest('DELETE', '/api/cache/clear');
    }

    /**
     * Test connection to the FIST API
     * 
     * @returns {Promise<boolean>} True if connection is successful
     */
    async testConnection() {
        try {
            const health = await this.getHealth();
            return ['healthy', 'warning'].includes(health.status);
        } catch {
            return false;
        }
    }
}

/**
 * Convenience functions for quick usage
 */

/**
 * Quick function to moderate a single piece of content
 * 
 * @param {string} content - Text content to moderate
 * @param {string} token - API token
 * @param {string} [baseUrl='http://localhost:8000'] - Base URL of the FIST API
 * @param {Object} [options] - Additional moderation options
 * @returns {Promise<ModerationResult>} Moderation result
 */
async function moderateContent(content, token, baseUrl = 'http://localhost:8000', options = {}) {
    const client = new FistClient({ token, baseUrl });
    return await client.moderateContent(content, options);
}

/**
 * Quick function to moderate multiple pieces of content
 * 
 * @param {string[]} contents - Array of text content to moderate
 * @param {string} token - API token
 * @param {string} [baseUrl='http://localhost:8000'] - Base URL of the FIST API
 * @param {Object} [options] - Additional moderation options
 * @returns {Promise<ModerationResult[]>} Array of moderation results
 */
async function moderateBatch(contents, token, baseUrl = 'http://localhost:8000', options = {}) {
    const client = new FistClient({ token, baseUrl });
    return await client.moderateBatch(contents, options);
}

// Export for different module systems
if (isNode) {
    // Node.js CommonJS
    module.exports = {
        FistClient,
        ModerationResult,
        BatchJob,
        FistClientError,
        AuthenticationError,
        APIError,
        moderateContent,
        moderateBatch
    };
} else {
    // Browser ES6 modules or global
    if (typeof window !== 'undefined') {
        window.FistClient = FistClient;
        window.moderateContent = moderateContent;
        window.moderateBatch = moderateBatch;
    }
}

// ES6 module export (for modern environments)
export {
    FistClient,
    ModerationResult,
    BatchJob,
    FistClientError,
    AuthenticationError,
    APIError,
    moderateContent,
    moderateBatch
};

// Example usage
if (isNode && require.main === module) {
    // Example usage when run directly in Node.js
    (async () => {
        const client = new FistClient({
            token: 'fist_your_token_here',
            baseUrl: 'http://localhost:8000'
        });

        try {
            // Test connection
            if (await client.testConnection()) {
                console.log('✅ Connected to FIST API successfully');

                // Single content moderation
                const result = await client.moderateContent('This is a test message');
                console.log(`Moderation result: ${result.finalDecision} (${result.inappropriateProbability}%)`);

                // Batch processing
                const contents = ['Message 1', 'Message 2', 'Message 3'];
                const results = await client.moderateBatch(contents);
                console.log(`Batch processed ${results.length} items`);

                // System health
                const health = await client.getHealth();
                console.log(`System health: ${health.status}`);

            } else {
                console.log('❌ Failed to connect to FIST API');
            }
        } catch (error) {
            console.error('Error:', error.message);
        }
    })();
}

File: client_libraries/python/fist_client.py
"""
FIST Content Moderation Python Client Library

A Python client library for easy integration with the FIST Content Moderation API.
Provides simple methods for content moderation, batch processing, and monitoring.

Usage:
    from fist_client import FistClient
    
    client = FistClient(
        token="fist_your_token_here",
        base_url="https://your-fist-api.com"
    )
    
    # Single content moderation
    result = client.moderate_content("Some text to moderate")
    
    # Batch processing
    results = client.moderate_batch(["text1", "text2", "text3"])
"""
import requests
import time
from typing import List, Dict, Any, Optional, Union
from dataclasses import dataclass
from datetime import datetime


@dataclass
class ModerationResult:
    """Result of content moderation."""
    moderation_id: str
    content_hash: str
    inappropriate_probability: int
    ai_reason: str
    final_decision: str  # A, R, or M
    reason: str
    word_count: int
    percentage_used: float
    created_at: str
    from_cache: bool = False


@dataclass
class BatchJob:
    """Batch processing job information."""
    job_id: str
    status: str
    total_items: int
    processed_items: int
    progress_percent: float
    created_at: str
    started_at: Optional[str] = None
    completed_at: Optional[str] = None
    background_task_id: Optional[str] = None


class FistClientError(Exception):
    """Base exception for FIST client errors."""
    pass


class AuthenticationError(FistClientError):
    """Authentication failed."""
    pass


class APIError(FistClientError):
    """API request failed."""
    def __init__(self, message: str, status_code: int = None, response_data: Dict = None):
        super().__init__(message)
        self.status_code = status_code
        self.response_data = response_data


class FistClient:
    """
    Python client for FIST Content Moderation API.
    
    Provides easy-to-use methods for content moderation, batch processing,
    and system monitoring with automatic error handling and retries.
    """
    
    def __init__(
        self, 
        token: str, 
        base_url: str = "http://localhost:8000",
        timeout: int = 30,
        max_retries: int = 3,
        retry_delay: float = 1.0
    ):
        """
        Initialize FIST client.
        
        Args:
            token: API token (should start with 'fist_')
            base_url: Base URL of the FIST API
            timeout: Request timeout in seconds
            max_retries: Maximum number of retries for failed requests
            retry_delay: Delay between retries in seconds
        """
        self.token = token
        self.base_url = base_url.rstrip('/')
        self.timeout = timeout
        self.max_retries = max_retries
        self.retry_delay = retry_delay
        
        # Validate token format
        if not token.startswith('fist_'):
            raise ValueError("Token must start with 'fist_'")
        
        self.session = requests.Session()
        self.session.headers.update({
            'Authorization': f'Bearer {token}',
            'Content-Type': 'application/json',
            'User-Agent': 'FIST-Python-Client/1.0.0'
        })
    
    def _make_request(
        self, 
        method: str, 
        endpoint: str, 
        data: Dict = None,
        params: Dict = None
    ) -> Dict[str, Any]:
        """Make HTTP request with retry logic."""
        url = f"{self.base_url}{endpoint}"
        
        for attempt in range(self.max_retries + 1):
            try:
                if method.upper() == 'GET':
                    response = self.session.get(url, params=params, timeout=self.timeout)
                elif method.upper() == 'POST':
                    response = self.session.post(url, json=data, timeout=self.timeout)
                elif method.upper() == 'DELETE':
                    response = self.session.delete(url, timeout=self.timeout)
                else:
                    raise ValueError(f"Unsupported HTTP method: {method}")
                
                # Handle authentication errors
                if response.status_code == 401:
                    raise AuthenticationError("Invalid or expired token")
                
                # Handle other client/server errors
                if response.status_code >= 400:
                    try:
                        error_data = response.json()
                        error_message = error_data.get('detail', f'HTTP {response.status_code}')
                    except:
                        error_message = f'HTTP {response.status_code}: {response.text}'
                    
                    raise APIError(
                        error_message, 
                        status_code=response.status_code,
                        response_data=error_data if 'error_data' in locals() else None
                    )
                
                return response.json()
                
            except (requests.exceptions.RequestException, APIError) as e:
                if attempt == self.max_retries:
                    raise
                
                if isinstance(e, APIError) and e.status_code < 500:
                    # Don't retry client errors (4xx)
                    raise
                
                time.sleep(self.retry_delay * (2 ** attempt))  # Exponential backoff
        
        raise APIError("Max retries exceeded")
    
    def moderate_content(
        self,
        content: str,
        percentages: Optional[List[float]] = None,
        thresholds: Optional[List[int]] = None,
        probability_thresholds: Optional[Dict[str, int]] = None
    ) -> ModerationResult:
        """
        Moderate a single piece of content.
        
        Args:
            content: Text content to moderate
            percentages: Custom percentages for content piercing
            thresholds: Custom word count thresholds
            probability_thresholds: Custom probability thresholds
            
        Returns:
            ModerationResult object with moderation details
        """
        if not content or not content.strip():
            raise ValueError("Content cannot be empty")
        
        data = {"content": content}
        if percentages is not None:
            data["percentages"] = percentages
        if thresholds is not None:
            data["thresholds"] = thresholds
        if probability_thresholds is not None:
            data["probability_thresholds"] = probability_thresholds
        
        response = self._make_request('POST', '/api/moderate', data=data)
        
        result_data = response['result']
        return ModerationResult(
            moderation_id=result_data['moderation_id'],
            content_hash=result_data['content_hash'],
            inappropriate_probability=result_data['ai_result']['inappropriate_probability'],
            ai_reason=result_data['ai_result']['reason'],
            final_decision=result_data['final_decision'],
            reason=result_data['reason'],
            word_count=result_data['word_count'],
            percentage_used=result_data['percentage_used'],
            created_at=result_data['created_at']
        )
    
    def moderate_batch(
        self,
        contents: List[str],
        percentages: Optional[List[float]] = None,
        thresholds: Optional[List[int]] = None,
        probability_thresholds: Optional[Dict[str, int]] = None,
        background: bool = False,
        poll_interval: float = 2.0,
        max_wait_time: int = 300
    ) -> Union[List[ModerationResult], BatchJob]:
        """
        Moderate multiple pieces of content in batch.
        
        Args:
            contents: List of text content to moderate
            percentages: Custom percentages for content piercing
            thresholds: Custom word count thresholds
            probability_thresholds: Custom probability thresholds
            background: Process in background (returns BatchJob for polling)
            poll_interval: Polling interval for background jobs (seconds)
            max_wait_time: Maximum time to wait for background jobs (seconds)
            
        Returns:
            List of ModerationResult objects (if background=False)
            or BatchJob object (if background=True)
        """
        if not contents:
            raise ValueError("Contents list cannot be empty")
        
        if len(contents) > 100:
            raise ValueError("Batch size cannot exceed 100 items")
        
        data = {
            "contents": contents,
            "background": background
        }
        if percentages is not None:
            data["percentages"] = percentages
        if thresholds is not None:
            data["thresholds"] = thresholds
        if probability_thresholds is not None:
            data["probability_thresholds"] = probability_thresholds
        
        response = self._make_request('POST', '/api/moderate/batch', data=data)
        
        if background:
            # Return BatchJob for polling
            batch_job = BatchJob(
                job_id=response['job_id'],
                status=response['status'],
                total_items=response['total_items'],
                processed_items=response['processed_items'],
                progress_percent=response['progress_percent'],
                created_at=response['created_at'],
                background_task_id=response.get('background_task_id')
            )
            
            # Wait for completion if requested
            if max_wait_time > 0:
                return self._wait_for_batch_completion(batch_job.job_id, poll_interval, max_wait_time)
            
            return batch_job
        else:
            # Return results directly
            results = []
            for result_data in response.get('results', []):
                results.append(ModerationResult(
                    moderation_id=result_data['moderation_id'],
                    content_hash=result_data['content_hash'],
                    inappropriate_probability=result_data['ai_result']['inappropriate_probability'],
                    ai_reason=result_data['ai_result']['reason'],
                    final_decision=result_data['final_decision'],
                    reason=result_data['reason'],
                    word_count=result_data['word_count'],
                    percentage_used=result_data['percentage_used'],
                    created_at=result_data['created_at'],
                    from_cache=result_data.get('from_cache', False)
                ))
            return results
    
    def get_batch_status(self, job_id: str) -> BatchJob:
        """
        Get status of a batch processing job.
        
        Args:
            job_id: Batch job ID
            
        Returns:
            BatchJob object with current status
        """
        response = self._make_request('GET', f'/api/moderate/batch/{job_id}/status')
        
        return BatchJob(
            job_id=response['job_id'],
            status=response['status'],
            total_items=response['total_items'],
            processed_items=response['processed_items'],
            progress_percent=response['progress_percent'],
            created_at=response.get('created_at', ''),
            started_at=response.get('started_at'),
            completed_at=response.get('completed_at')
        )
    
    def get_batch_results(self, job_id: str) -> List[ModerationResult]:
        """
        Get results of a completed batch job.
        
        Args:
            job_id: Batch job ID
            
        Returns:
            List of ModerationResult objects
        """
        response = self._make_request('GET', f'/api/moderate/batch/{job_id}/results')
        
        results = []
        for result_data in response.get('results', []):
            results.append(ModerationResult(
                moderation_id=result_data['moderation_id'],
                content_hash=result_data['content_hash'],
                inappropriate_probability=result_data['ai_result']['inappropriate_probability'],
                ai_reason=result_data['ai_result']['reason'],
                final_decision=result_data['final_decision'],
                reason=result_data['reason'],
                word_count=result_data['word_count'],
                percentage_used=result_data['percentage_used'],
                created_at=result_data['created_at'],
                from_cache=result_data.get('from_cache', False)
            ))
        return results
    
    def _wait_for_batch_completion(
        self, 
        job_id: str, 
        poll_interval: float, 
        max_wait_time: int
    ) -> List[ModerationResult]:
        """Wait for batch job completion and return results."""
        start_time = time.time()
        
        while time.time() - start_time < max_wait_time:
            status = self.get_batch_status(job_id)
            
            if status.status == 'completed':
                return self.get_batch_results(job_id)
            elif status.status == 'failed':
                raise APIError(f"Batch job {job_id} failed")
            
            time.sleep(poll_interval)
        
        raise APIError(f"Batch job {job_id} did not complete within {max_wait_time} seconds")
    
    def get_health(self) -> Dict[str, Any]:
        """
        Get system health status.
        
        Returns:
            Dictionary with health check results
        """
        return self._make_request('GET', '/api/health')
    
    def get_metrics(self) -> Dict[str, Any]:
        """
        Get system performance metrics.
        
        Returns:
            Dictionary with performance metrics
        """
        return self._make_request('GET', '/api/metrics')
    
    def get_cache_stats(self) -> Dict[str, Any]:
        """
        Get cache performance statistics.
        
        Returns:
            Dictionary with cache statistics
        """
        return self._make_request('GET', '/api/cache/stats')
    
    def clear_cache(self) -> Dict[str, Any]:
        """
        Clear the system cache.
        
        Returns:
            Dictionary with clear operation results
        """
        return self._make_request('DELETE', '/api/cache/clear')
    
    def test_connection(self) -> bool:
        """
        Test connection to the FIST API.
        
        Returns:
            True if connection is successful, False otherwise
        """
        try:
            health = self.get_health()
            return health.get('status') in ['healthy', 'warning']
        except:
            return False


# Convenience functions for quick usage
def moderate_content(
    content: str, 
    token: str, 
    base_url: str = "http://localhost:8000",
    **kwargs
) -> ModerationResult:
    """
    Quick function to moderate a single piece of content.
    
    Args:
        content: Text content to moderate
        token: API token
        base_url: Base URL of the FIST API
        **kwargs: Additional arguments for moderation
        
    Returns:
        ModerationResult object
    """
    client = FistClient(token=token, base_url=base_url)
    return client.moderate_content(content, **kwargs)


def moderate_batch(
    contents: List[str], 
    token: str, 
    base_url: str = "http://localhost:8000",
    **kwargs
) -> List[ModerationResult]:
    """
    Quick function to moderate multiple pieces of content.
    
    Args:
        contents: List of text content to moderate
        token: API token
        base_url: Base URL of the FIST API
        **kwargs: Additional arguments for moderation
        
    Returns:
        List of ModerationResult objects
    """
    client = FistClient(token=token, base_url=base_url)
    return client.moderate_batch(contents, **kwargs)


# Example usage
if __name__ == "__main__":
    # Example usage of the FIST client
    client = FistClient(
        token="fist_your_token_here",
        base_url="http://localhost:8000"
    )
    
    # Test connection
    if client.test_connection():
        print("✅ Connected to FIST API successfully")
        
        # Single content moderation
        result = client.moderate_content("This is a test message")
        print(f"Moderation result: {result.final_decision} ({result.inappropriate_probability}%)")
        
        # Batch processing
        contents = ["Message 1", "Message 2", "Message 3"]
        results = client.moderate_batch(contents)
        print(f"Batch processed {len(results)} items")
        
        # System health
        health = client.get_health()
        print(f"System health: {health['status']}")
        
    else:
        print("❌ Failed to connect to FIST API")
